### DISCUSSION TITLE
Segmentation은 왜 필요했을까요?

### DISCUSSION THREAD

**Discussion by mypark42** [exciting-kfs/kfs#13] [General]
제가 생각한 이유는 

1. 어드레스 확장.
2. 각 프로세스 고립을 통해 보안 기능 구현.

인데, 다른 목적이 또 있었을까요?

**Comment by cjeon42** [comment]
확실히 보안 / 안정성이 주 목적인 것 같아요.

말씀하신 것 외에 추가적으로 Read / Write / Execute 권한을 세그먼트마다 정해서

stack과 관련된 세그먼트에는 read / write 만 주고
code와 관련된 세그먼트에는 read / execute 만 주는 식으로 더 강력한 보안을 구현할 수 있는 것 같습니다.
